notebook/static/notebook/js/notebook.js

```diff
@@ -283,6 +292,10 @@ define(function (require) {
             if (kill_kernel) {
                 that.session.delete();
             }
+            // if the kernel is busy, prompt the user if he’s sure
+            if (that.kernel_busy) {
+                return "The Kernel is busy, changes may be lost.";
+            }
```

Comment:
I think this should be after the unsaved changes check - if there are unsaved changes and the kernel is busy, it's probably the unsaved changes that are more important.

I'd also tweak the wording of the message a bit: "... *outputs* may be lost".